From pingc@wacom.com  Thu Sep 28 16:49:52 2006
From: Eric Sesterhenn [mailto:snakebyte@gmx.de]
Sent: Monday, September 18, 2006 3:34 PM
To: kernel-janitors@lists.osdl.org
Subject: USB: BUG_ON conversion for wacom.c

this patch converts two if () BUG(); construct to BUG_ON();
which occupies less space, uses unlikely and is safer when
BUG() is disabled.

Signed-off-by: Eric Sesterhenn <snakebyte@gmx.de>
Acked-by: "Ping Cheng" <pingc@wacom.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/input/wacom_sys.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/usb/input/wacom_sys.c
+++ gregkh-2.6/drivers/usb/input/wacom_sys.c
@@ -230,8 +230,7 @@ static int wacom_probe(struct usb_interf
 	strlcat(wacom->phys, "/input0", sizeof(wacom->phys));
 
 	wacom_wac->features = get_wacom_feature(id);
-	if (wacom_wac->features->pktlen > 10)
-		BUG();
+	BUG_ON(wacom_wac->features->pktlen > 10);
 
 	input_dev->name = wacom_wac->features->name;
 	wacom->wacom_wac = wacom_wac;
